### ISSUE TITLE
cannot set a socket into a non-blocking mode

### ISSUE THREAD

**Issue by jvo203** [neachdainn/nng-rs#5]
The "set_nonblocking" method is not recognised:

error[E0599]: no method named `set_nonblocking` found for type `nng::socket::Socket` in the current scope
    --> src/main.rs:3028:36
     |
3028 |                             socket.set_nonblocking(true);

The socket is initialised in the code below. Re-arranging the code in various ways, adding "mut socket" etc. does not seem to solve the problem... The nng version used is 0.5.2.

let addr = format!("tcp://0.0.0.0:{}", port);
                    println!("nanomsg-next server address: {}", addr);

                    let socket = nng::Socket::new(nng::Protocol::Rep0);
                    match socket {
                        Ok(socket) => {
                            socket.set_nonblocking(true);
                            let endpoint = socket.listen(&addr);
                            match endpoint {
                                Ok(_) => {
                                    nn_server[i] = Some(socket);
                                    nn_port[i] = Some(port);
                                }
                                Err(err) => println!("nanomsg-next listen error: {}", err),
                            }
                        }
                        Err(err) => println!("nanomsg-next protocol error: {}", err),
                    };

**Comment by neachdainn** [comment]
The `Socket::set_nonblocking` function was removed in v0.5.0 (see the [changelog][1] and the [Gitlab Issue][2]). I'm also not sure what you mean by v0.5.2 as that version does not exist.

[1]: https://github.com/neachdainn/nng-rs/blob/master/CHANGELOG.adoc#removed-1
[2]: https://gitlab.com/neachdainn/nng-rs/issues/34

**Comment by jvo203** [comment]
Hi, Thank you for a prompt reply. Version numbers: prior to nng-rs I had been using Rust bindings to nanomsg, which are at 0.7.2. So just before going to bed (late night in Japan) I must have taken v0.5 from your crate and .2 from the previously used nanomsg.

Anyway, it seems that for many Rust crates Google serves old versions of documentation. Googling for "nng::Socket Rust" brings the documentation for the old very old version 0.2.1 as the top result... On the other hand, searching for "nng Rust" takes one to the latest docs for 0.5.0.

**Comment by jvo203** [comment]
Got Socket.recv() and try_recv() to work. Unfortunately the last message sent from a client always gets lost. It gets sent on the client side (.send() returns Ok()) but the server never receives it. It happens all the time, as if there was a need to flush the client after the last message. All the messages prior to the last one always get through. But the last one never does. Reproducible every time...

Because of this critical issue I have decided to return to ZeroMQ Rust bindings (at least until a 100% reliable library can be found). Last week I started with ZeroMQ (Rust libzmq) but got issues to do with multithreading. Then went to Rust nanomsg but could not get it to work in a threaded code. So jumped into nng (nanomsg-next). It is as easy as using libzmq but unfortunately there seems to be this show-stopper with the last message never getting through.

This last message can be fairly large (anything between 200KB and 200MB). Previous (successfully delivered) messages are fairly small (2000 bytes etc). ZeroMQ had no problems whatsoever delivering all messages. Life is full of difficult choices!
